xrootd/encoder/encoder.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package encoder // import "go-hep.org/x/hep/xrootd/encoder"
+
+import (
+	"encoding/binary"
+	"reflect"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/protocol"
+)
+
+// MarshalRequest encodes request body alongside with request and stream ids
+func MarshalRequest(requestID uint16, streamID protocol.StreamID, requestBody interface{}) ([]byte, error) {
+	requestHeader := make([]byte, 4)
+
+	copy(requestHeader, streamID[:])
+	binary.BigEndian.PutUint16(requestHeader[2:], requestID)
+
+	b, err := Marshal(requestBody)
+	if err != nil {
+		return nil, err
+	}
+
+	return append(requestHeader, b...), nil
+}
+
+// Marshal encodes structure to the bytes following the XRootd protocol specification.
```

Comment:
`s/Marshal encodes structure to the bytes/Marshal encodes a structure value into a slice of bytes/`